src/hooks/useConceptExerciseIssues.ts

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useState } from 'react'
+
+export interface ConceptExerciseIssue {
+  number: number
+  title: string
+  body: string
+  url: string
+  updatedAt: string
+}
+
+export type ConceptExerciseIssueResult<T> = {
+  result: T | undefined
+  error: boolean
+  done: boolean
+}
+
+function useConceptExerciseIssuesApi<T>(
+  url: string
+): ConceptExerciseIssueResult<T> {
+  const [state, setState] = useState<ConceptExerciseIssueResult<T>>({
+    result: undefined,
+    error: false,
+    done: false,
+  })
+
+  useEffect(() => {
+    if (state.done === true) {
+      return
+    }
+
+    let isMounted = true
```

Comment:
Technically incorrect. Sure, the _hook_ won't be mounted, but the component might be. This hook _reruns_ each time `url` and `state` change. I usually use:

```typescript
let stillCareAboutThis = true
```

But I've had random feedback about that being a bad name (without _any_ explanation why). So perhaps `let requestIsStale = false` and flip it to true when the hook's cleanup runs.